R/cell_count_plot.R

```diff
@@ -59,41 +60,57 @@ CellCountPlot.data.frame <- function(
     gg <- object %>%
       group_by(.data[[group_by]], .data[[color_by]]) %>%
       count() %>%
+      ungroup() %>%
+      group_by(!!sym(group_by)) %>%
+      mutate(frequency = n / sum(n) * 100) %>%
       ungroup()
-    nudge_y <- max(gg$n) / 20
-    p <- gg %>%
-      ggplot(aes(.data[[group_by]], n, fill = .data[[color_by]])) +
-      geom_col(position = position_dodge(width = 0.95)) +
+
+    nudge_y <- if(as_frequency) max(gg$frequency) / 20 else max(gg$n) / 20
+
+    p <- ggplot(gg, aes(x = .data[[group_by]],
+                        y = if(as_frequency) frequency else n,
+                        fill = .data[[color_by]])) +
+      geom_col(position = if(stack) "stack" else position_dodge(width = 0.95)) +
       {
         if (show_count) {
-          geom_text(aes(.data[[group_by]], n + nudge_y,
-                        label = n,
+          geom_text(aes(label = if(as_frequency) sprintf("%.1f%%", frequency) else n,
                         group = .data[[color_by]]),
-            position = position_dodge(width = 1)
-          )
+                    position = if(stack) position_stack(vjust = 0.5) else position_dodge(width = 0.95),
+                    size = 3)
         }
       } +
-      labs(title = paste0("Cell counts for ", group_by,
-                          " colored by ", color_by))
+      labs(title = ifelse(as_frequency,
+                       paste0("Cell frequencies for ", group_by, " colored by ", color_by),
+                       paste0("Cell counts for ", group_by, " colored by ", color_by)),
+        y = ifelse(as_frequency, "Frequency (%)", "Count"))
   } else {
     gg <- object %>%
       group_by(.data[[color_by]]) %>%
       count() %>%
-      ungroup()
-    nudge_y <- max(gg$n) / 20
-    p <- gg %>%
-      ggplot(aes(.data[[color_by]], n, fill = .data[[color_by]])) +
+      ungroup() %>%
+      mutate(frequency = n / sum(n) * 100)
+
+    nudge_y <- if(as_frequency) max(gg$frequency) / 20 else max(gg$n) / 20
+
+    p <- ggplot(gg, aes(x = .data[[color_by]],
+                        y = if(as_frequency) frequency else n,
+                        fill = .data[[color_by]])) +
       geom_col(position = position_dodge(width = 0.95)) +
       {
         if (show_count) {
-          geom_text(aes(.data[[color_by]], n + nudge_y, label = n))
+          geom_text(aes(label = if(as_frequency) sprintf("%.1f%%", frequency) else n),
```

Comment:
The variable `frequency` needs to be added to `globalVariables(...)` in the `aaa.R` script for the package check. You can see the notes here:  https://github.com/PixelgenTechnologies/pixelatorR/actions/runs/10252352635/job/28362538766?pr=52

Reply:
I have added `frequency` as a global variable.